main.go

```diff
@@ -34,7 +34,9 @@ func init() {
 var (
 	listenAddress       = flag.String("web.listen-address", ":9102", "The address on which to expose the web interface and generated Prometheus metrics.")
 	metricsEndpoint     = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
-	statsdListenAddress = flag.String("statsd.listen-address", ":9125", "The UDP address on which to receive statsd metric lines.")
+	statsdListenAddress = flag.String("statsd.listen-address", ":9125", "The UDP/TCP address on which to receive statsd metric lines.")
+	statsdListenUDP     = flag.Bool("statsd.listen-udp", true, "Whether to receive UDP statsd metrics.")
```

Comment:
Instead of boolean flags and introducing a dependency between flags, it'd be simpler to just have to flags with the addresses for each protocol, like `-statsd.listen-address-udp=":9125"` `-statsd.listen-address-tcp=":9125"`. An empty value signals disables the server on that protocol.